ui/StatusQ/include/StatusQ/writableproxymodel.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <QAbstractProxyModel>
+
+// WritableProxyModel is a QAbstractProxyModel that allows you to modify the data without modifying the source model.
+// It is useful for implementing a "dirty" state for a model, where you can modify the data and then commit the changes
+// to the source model.
+
+// Supported features (reimplemented):
+// - setData
+// - setItemData
+// - removeRows
+// - insertRows
+// - moveRows (TODO)
+// - toVariantMap
+// - to be continued...
+
+// Limitations:
+// - only 1 column (no grid models)
+// - no parent index (no tree models)
+
+class WritableProxyModelPrivate;
+
+class WritableProxyModel : public QAbstractProxyModel
```

Comment:
`QAbstractProxyModel` is probably ok as well, but I would consider also `QAbstractListModel` as a base model. It gives `columnCount`, `index`, `sibling`, `parent` out of the box.

Reply:
We need a `QAbstractProxyModel` derivate. `QAbstractListModel` won't help.